File: 2018/solutions/prob06.cpp
#include <iostream>
#include <string>

int checkdigit(const std::string &number);
char digitOrX(int digit);

int main() {
  std::string number;

  while (std::cin >> number) {
    std::cout << number << digitOrX(checkdigit(number)) << std::endl;
  }

  return EXIT_SUCCESS;
}


int checkdigit(const std::string &number)
{
  int sum = 0;
  for (int i = 0; i < 9; i++) {
    sum += (number[i] - '0') * (i + 1);
  }

  return sum % 11;
}

char digitOrX(int digit)
{
  if (digit == 10)
    return 'X';
  return (char) digit + '0';
}

File: 2018/solutions/prob06.java
import java.io.*;

public class prob06 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader
                = new BufferedReader(new InputStreamReader(System.in));

        String line = null;
        while ((line = reader.readLine()) != null) {
            String number = line;
            System.out.printf("%s%s%n", number, digitOrX(checkdigit(number)));
        }
    }

    public static int checkdigit(String number) {
        int sum = 0;
        for (int i = 0; i < 9; i++) {
            sum += (number.charAt(i) - '0') * (i + 1);
        }

        return sum % 11;
    }

    public static String digitOrX(int digit) {
        if (digit == 10)
            return "X";
        else
            return Integer.toString(digit);
    }
}
